homeassistant/helpers/config_validation.py

```diff
@@ -267,6 +297,19 @@ def url(value: Any) -> str:
     raise vol.Invalid('invalid url')
 
 
+def email(value: Any) -> str:
+    """Validate an email address."""
+    email_in = str(value)
+    email_re = re.match(
+        r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)",
```

Comment:
Are you saying we should allow localhost? pretty sure that regex won't pass for localhost/local DNS entries